python/cudf/cudf/core/column/categorical.py

```diff
@@ -1472,6 +1463,53 @@ def view(self, dtype: Dtype) -> ColumnBase:
             "Categorical column views are not currently supported"
         )
 
+    @staticmethod
+    def _concat(objs: MutableSequence[CategoricalColumn]) -> CategoricalColumn:
+        # TODO: This function currently assumes it is being called from
+        # column._concat_columns, at least to the extent that all the
+        # preprocessing in that function has already been done. That should be
+        # improved as the concatenation API is solidified.
+
+        # Find the first non-null column:
+        head = next((obj for obj in objs if obj.valid_count), objs[0])
+
+        # Combine and de-dupe the categories
+        cats = (
+            cudf.concat([o.cat().categories for o in objs])
+            .to_series()
+            .drop_duplicates(ignore_index=True)
+            ._column
+        )
+        objs = [
+            o.cat()._set_categories(o.cat().categories, cats, is_unique=True)
+            for o in objs
+        ]
+        # Map `objs` into a list of the codes until we port Categorical to
+        # use the libcudf++ Category data type.
```

Comment:
You mean dictionary type?

Reply:
Actually this comment is a holdover from before. I wasn't sure what it referred to and didn't prioritize figuring that out. I'm not sure what this is meant to say.

Reply:
As far as I know there isn't a Category type in libcudf, but there is a dictionary type is more closely aligned with the memory layout of Category type.

Reply:
In Python the `category` dtype isn't hooked up to any libcudf type. When handing a categorical column down, we get only the codes, hand that to libcudf as an integer column, and then reattach the categories when we receive a result from libcudf.

Reply:
Right, so that sounds like this comment is probably outdated at best and can safely be removed in this PR.